sys/newlib: Bugfix, add missing syscalls (only stubs)

jcarrano opened the pull request:
Add small stubs for _times and _link_r so that code using times() or link()/rename() can still compile. The functions are marked as not implemented and return invalid codes.

### Issues/PRs references

Fixes #9051 .

kaspar030 commented on sys/newlib_syscalls_default/syscalls.c:
```diff
@@ -516,3 +535,17 @@ int _gettimeofday_r(struct _reent *r, struct timeval *restrict tp, void *restric
+clock_t _times(struct tms *ptms)
+{
+    memset(ptms, 0, sizeof(*ptms));
+
```
does errno have to be cleared? Spec says, errno indicates error on error...

jcarrano replied:
@kaspar030 You are right, this is an example from libgloss (part of newlib):
```
/* _times -- no clock, so return an error.  */
clock_t _times (struct tms *buf)
{
  errno = EINVAL;
  return (-1);
}
```
I'm fixing it.

kaspar030 commented:
please squash

basilfx commented:
@jcarrano The build failed, because of unused parameters.

```
syscalls.c: In function '_times':
syscalls.c:546:28: error: unused parameter 'ptms' [-Werror=unused-parameter]
 clock_t _times(struct tms *ptms)
```

You can mark them as unused by adding `(void) ptms;` to `_times`.

kaspar030 commented on sys/newlib_syscalls_default/syscalls.c:
```diff
@@ -516,3 +535,18 @@ int _gettimeofday_r(struct _reent *r, struct timeval *restrict tp, void *restric
+{
+    (void)ptms;
+
+    errno = EINVAL;
```
Shouldn't this also use the errno variable from struct reent?

jcarrano replied:
I copied it from libgloss, which is part of newlib.

jia200x commented:
@jcarrano any news on this?

jcarrano commented:
@jia200x I addressed @kaspar030 review, and squashed. Btw, I'm also working intermittently on what I think is a better fix (#9258  ), but this one should work for now.

tcschmidt commented:
@kaspar030 do changes satisfy your concerns?

danpetry commented:
OK I'm satisfied with the approach here, on the basis that it replicates the approach elsewhere and that that approach is being addressed more comprehensively in https://github.com/RIOT-OS/RIOT/pull/9258. Also, the added functions aren't actually used elsewhere meaning, for example, they won't have a memory impact anywhere except in the Lua interpreter. Would be good to address the above PR ASAP to avoid errors if other people try to use the related newlib functions.
